addons/ui/fnc_openItemContextMenu.sqf

```diff
@@ -0,0 +1,192 @@
+#include "script_component.hpp"
+/* ----------------------------------------------------------------------------
+Internal Function: cba_ui_fnc_openItemContextMenu
+
+Description:
+    Opens the item context menu on the inventory screen.
+
+Parameters:
+    _display - RscDisplayInventory display <DISPLAY>
+    _item    - Item classname <STRING>
+
+Returns:
+    Nothing/Undefined.
+
+Examples:
+    (begin example)
+        call cba_ui_fnc_openItemContextMenu;
+    (end)
+
+Author:
+    commy2
+---------------------------------------------------------------------------- */
+
+params ["_display", "_container", "_item", "_slot"];
+if (_item isEqualTo "") exitWith {};
+
+// ---
+// Read context menu options.
+private _config = _item call CBA_fnc_getItemConfig;
+private _superclass = configName (configHierarchy _config param [1, configNull]);
+
+private _options = [];
+while {
+    _options pushBack (GVAR(ItemContextMenuOptions) getVariable configName _config);
+    _config = inheritsFrom _config;
+    !isNull _config
+} do {};
+
+switch (toUpper _superclass) do {
+    // magazines
+    case "CFGMAGAZINES": {
+        _options pushBack (GVAR(ItemContextMenuOptions) getVariable "#AllMagazines");
+    };
+
+    // Other items, weapons
+    case "CFGGLASSES";
+    case "CFGWEAPONS": {
+        _options pushBack (GVAR(ItemContextMenuOptions) getVariable "#AllItems");
+    };
+};
+
+_options pushBack (GVAR(ItemContextMenuOptions) getVariable "#All");
+
+// Skip menu if no options.
+if (count _options isEqualTo 0) exitWith {};
+
+// ---
+// Create context menu.
+// ctrlSetBackgroundColor command does not seem to work for RscListBox.
+private _list = _display ctrlCreate [QGVAR(ItemContextMenu), -1];
+
+// ---
+// Populate context menu with options.
+{
+    _x params ["_slots", "_displayName", "_tooltip", "_color", "_icon", "_conditionEnable", "_conditionShow", "_statement", "_consume", "_params"];
+
+    private _args = [_unit, _container, _item, _slot, _params];
+    if (isLocalized _displayName) then {
+        _displayName = localize _displayName;
+    };
+
+    if (isLocalized _tooltip) then {
+        _tooltip = localize _tooltip;
+    };
+
+    if ((_slot in _slots || {"ALL" in _slots}) && {_args call _conditionShow}) then {
+        private _index = _list lbAdd _displayName;
+        _list lbSetTooltip [_index, _tooltip]; // broken, @todo
+
+        private _key = format [QGVAR(OptionData%1), _index];
+
+        if (_color isEqualTo []) then {
+            _color = getArray (configFile >> ctrlClassName _list >> "colorText");
+        };
+
+        _list lbSetColor [_index, _color];
+        _list lbSetData [_index, _key];
+        _list lbSetPicture [_index, _icon];
+
+        if !([_args call _conditionEnable] param [0, false]) then {
```

Comment:
Vulnerable to assignment return. Irrelevant for conditions?
Comment should describe that this treats nil as then case instead of neither then nor else.